build_script/publications.gradle

```diff
@@ -61,13 +61,19 @@ def manipulatePomXml = {
 publishing {
   publications {
     // Define the "release" publication
-    release(MavenPublication) {
-      from components.java
+    release(MavenPublication) { publication ->
+      if (project.name == 'restli-netty') {
+        apply plugin: 'com.github.johnrengelman.shadow'
+        project.shadow.component(publication)
+      } else {
+        from components.java
+      }
 
       afterEvaluate {
         // Add all extra archives (sources, javadoc, any custom archives e.g. all)
         project.configurations.archives.allArtifacts.forEach {
-          if (it.classifier) {
+          // Skip 'slim' identifier, otherwise this logic attemps to publish the disabled slim jar for restli-netty.
+          if (it.classifier && it.classifier != 'slim') {
```

Comment:
What exact JARs will this new module publish and with what exact classifiers? This is not clear looking at the code.

Reply:
The restli-netty jar is published with classifier shadow (the standard classifier for the shaded jar with this plugin). The "slim" seen here is the default jar task in restli-netty.

The details on why it is here and the classifier for the default jar task was set to "slim" can be found in build.gradle for restli-netty. There is a longstanding bug in IntelliJ when using the shadow plugin; IntelliJ will still think it should use the classpaths as it would determine from analyzing the default jar task. This means after shading Netty, while the builds all work, IntelliJ will give everyone spurious errors in their code that uses shaded Netty classes. The workaround is to assign a classifier to the default jar task. This issue is being tracked as IDEA-163411 on their Youtrack.

Reply:
Thanks for the explanation. Could you leave a TODO in the code here so that in the future, the context isn't totally lost on us?

Reply:
Done.